### PR TITLE
fix(datepicker): month selection

### LINKED ISSUE #2128: [Datepicker] Fix behavior of MonthYearSelect and NextMonthButton with multiple months
In Calendar with multiple months, the MonthYearSelect and NextMonthButton of 2nd month is not working correctly. 

MonthYearSelect: If October is selected in the 2nd month, the 2nd month will be November. And the 1st month will be October

NextMonthButton: If NextMonthButton is clicked, the months are increased by 2

And when `orientation={ORIENTATION.horizontal}` is applied, the MonthYearSelect is not shown. But without orientation prop, MonthYearSelect is shown. So I think the author decided to not support MonthYearSelect for multiple months and hide it but forgot to handle when orientation is undefined.

How about to support MonthYearSelect with multiple months too? It's not that hard and users who didn't set oritentation prop alerady see MonthYearSelect.

https://codesandbox.io/s/calendar-with-multiple-months-1cn1l
~https://codesandbox.io/s/calendar-with-multiple-months-d3rgw~

## Current Behavior

MonthYearSelect: If August is selected in the 2nd month, the 2nd month will be September. And the 1st month will be August.

baseui-datepicker-month-select-bug

NextMonthButton: If NextMonthButton is clicked, the months are increased by 2

baseui-datepicker-next-button-bug

## Expected Behavior

MonthYearSelect: If August is selected in the 2nd month, the 2nd month will be August. And the 1st month will be July.

NextMonthButton: If NextMonthButton is clicked, the months are increased by 1

## Your Environment

| Tech    | Version |
| ------- | ------- |
| Base UI | v9.9.0  |
| React   |         |
| browser |  Firefox 70.0b14 |

- [x] I have searched the [issues](https://github.com/uber/baseweb/issues) of this repository and believe that this is not a duplicate.

### PR DESCRIPTION
**Pull request by lhbrennan**
Fixes #2128

#### Description

Previously, in a multi-month display:
* the 'previous' arrow button went back by 1 month, but the 'next'
button went forward by 1 + the number of months displayed
* vertical orientation did not work at all
* when orientation was undefined, the fallback value was 'vertical' (it
is now 'horizontal')

#### Scope
Patch: Bug Fix

### CODE REVIEW

File: src/datepicker/calendar-header.js [hunk 1]
```diff
@@ -86,15 +86,6 @@ export default class CalendarHeader<T = Date> extends React.Component<
     return this.props.date || this.dateHelpers.date();
   };
```
Comment (lhbrennan) [inline src/datepicker/calendar-header.js:88]: `handleMonthChange` seems to not be used anywhere ¯\_(ツ)_/¯

File: src/datepicker/styled-components.js [hunk 2]
```diff
@@ -36,7 +37,15 @@ export const StyledRoot = styled<SharedStylePropsT>('div', props => {
   };
 });
 
-export const StyledMonthContainer = styled('div', {display: 'flex'});
+export const StyledMonthContainer = styled<{
+  $orientation: $PropertyType<CalendarPropsT<Date>, 'orientation'>,
```
Comment (lhbrennan) [inline src/datepicker/styled-components.js:41]: is there a better way to type `$orientation`?

Comment (chasestarr) [reply src/datepicker/styled-components.js:41]: This is fine with me. Another option may be to pull orientation type definition out to its own type and import that to this file
https://github.com/uber/baseweb/blob/master/src/datepicker/types.js#L177

**Review by chasestarr — APPROVED** [review]
Awesome! This is working well. It's also neat that codesandbox extension included the sandbox from the issue here https://github.com/uber/baseweb/blob/master/src/datepicker/types.js#L177 Made it really easy to quickly verify that everything is working well

**Pull request merged**